lib/src/utils.dart

```diff
@@ -59,16 +59,37 @@ extension BoolValueUtils on bool {
 
 extension ListValueUtils<T extends ValueType> on List<T> {
   Value toValue() {
-    return Value(listValue: ListValue(values: this.map((e) => e.toValue())));
+    return Value(listValue: ListValue(values: map((e) => e.toValue())));
   }
 }
 
 extension MapStructUtils on Map<String, dynamic> {
   Struct toStruct({bool skipErrors = false}) {
     Map<String, Value> result = {};
-    for (var entry in this.entries) {
+    for (var entry in entries) {
       try {
-        final v = entry.value.toValue();
+        var value = entry.value;
+        // ignore: prefer_typing_uninitialized_variables
+        var v;
+        // ignore: type_check_with_null, prefer_void_to_null
+        if (value is Null) {
+          v = value.toValue();
+        }
+        if (value is double) {
+          v = value.toValue();
+        }
+        if (value is String) {
+          v = value.toValue();
+        }
+        if (value is bool) {
+          v = value.toValue();
+        }
+        if (value is List<ValueType>) {
+          v = value.toValue();
+        }
+        if (value is Map<String, dynamic>) {
+          v = value.toValue();
+        }
```

Comment:
Extensions get set at compile time, while the way we were typing dynamics is a runtime decision, so it was unable to find `toValue()` for any of the types and would throw an error. We need to get the type at runtime so we know which version of toValue to call.

I don't like this, but it works, it would crash if a type other than the ones checked for was inside the map, so that needs to be accounted for.

reference: 
https://stackoverflow.com/questions/61921608/dart-use-generic-extension-method-get-nosuchmethoderror-class-xxx-has-no-in

Reply:
it is confusing, i could probably explain it better too.

Here is the error it shows without having this type checking:

<img width="654" alt="Screenshot 2023-05-22 at 11 32 38 AM" src="https://github.com/viamrobotics/viam-flutter-sdk/assets/22106496/c916420a-b239-4968-8adc-c0f6d805e59c">

happy to try and explain offline

Reply:
I think I understand better after an offline explanation. Definitely don't love the list of `ifs` here that do a sort of implicit casting to avoid the runtime complaints, but it doesn't seem like there's an obvious alternative given the static nature of extensions. Could you leave a comment in this method for posterity as to why this function is set up the way it is? And, can we only catch `NoSuchMethodError`s instead of all exceptions?

Reply:
Benji and I spoke offline, since removing the `toValue()` calls for each type and calling the constructor directly in the Map method, we believe a comment isn't needed.

The only reason we'll catch the error is to log it, and then rethrow. so we don't need to target a specific error type.